proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1071 @@
+[LEDS](https://github.com/envoyproxy/envoy/issues/10373).  Rather than living
+with an arbitrary mix of opaque and structured forms, the URI-based xDS resource
+naming scheme below provides a structured scheme that can serve the
+expression needs of resource types such as VHDS and LEDS.
+
+#### Aliasing
+
+VHDS introduced aliasing to the xDS transport, since the same virtual host
+resource may be known by multiple names, e.g. `routeConfigA/foo.com,
+routeConfigA/bar.com`. This mechanism is not widely used yet elsewhere in xDS
+and it is another example of mixing the transport and data model resource layers
+in xDS.
+
+We introduce a redirection mechanism below that allows a control plane to
+provide aliasing semantics without requiring explicit transport-level support
+for aliasing. When querying for either `routeConfigA/foo.com` or `routeConfigA/bar.com`, the
+management server might issue a redirect causing the xDS client to fetch
+`someCanonicalVirtualHost` resource.
+
+This same mechanism is also useful when supporting resource delegation during
+federation and removes special case treatment of aliasing from the xDS transport
+protocol.
+
+### Related Proposals:
+
+* The original Google Docs design documents:
+  [Full](https://docs.google.com/document/d/1zZav-IYxMO0mP2A7y5XHBa9v0eXyirw1CxrLLSUqWR8/edit#heading=h.w8hw2zbv3jwl)
+  [Summary](https://docs.google.com/document/d/1m5_Q9LUlzvDdImP0jqh1lSTrLMldsApTQX6ibbd3i7c/edit?ts=5ec2fd8a#heading=h.w8hw2zbv3jwl)
+* [xDS transport protocol.](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol)
+
+## Proposal
+
+The proposal below is fully backwards compatible with the existing xDS transport
+protocol and does not require any changes to clients or servers that do not wish
+to opt into the new naming scheme.
+
+### URI-based xDS resource names
+
+xDS URIs follow a [RFC-3986](https://tools.ietf.org/html/rfc3986) compliant
+schema. The following schema is proposed for xDS resource names:
+
+`xdstp://[{authority}]/{resource type}/{id/*}?{context parameters}{#processing directive,*}`
+
+* `authority` is an opaque string naming a resource authority, e.g.
+  `some.control.plane`.
+* `resource type` is the xDS resource type URL, e.g. `envoy.config.route.v3.RouteConfiguration`
+* `id/*` is the remaining path component of the URI and is fully opaque; naming
+  is at the discretion of the control plane(s).
+* `context parameters` are the URI query parameters and express contextual
+  information for selecting resource variants, for example `shard_id=1234&direction=inbound`.
+* A `processing directive` provides additional information to the client on how
+  the URI is to be interpreted. Two supported directives are:
+  * `alt=<xdstp:// URI>` expressing a fallback URI.
+  * `entry=<resource name>` providing an anchor referring to an inlined resource
+    for [list collections](#list). Resource names must be of the form `[a-zA-Z0-9_-\./]+`.
```

Comment:
Can more than one _processing directive_ appear in a locator?
If so, can the same directive appear more than once?

Reply:
Yes on > 1 processing directive, see next sentence. I've clarified on the same directive multiple times.